crates/librqbit/src/session.rs

```diff
@@ -1365,8 +1365,20 @@ impl Session {
     }
 
     pub async fn pause(&self, handle: &ManagedTorrentHandle) -> anyhow::Result<()> {
-        handle.locked.write().paused = true;
-        handle.pause()?;
+        let prev_state;
```

Comment:
Can it be simplified by first calling handle.pause(), and then only if it was successful set paused = true?

This way the lock is taken only once

Reply:
Yeah, it makes sense. Updated.